File: javascript/etl/etl.js
export const transform = (data) => {
	let dataMap = new Map(Object.entries(data));
	let transformedMap = new Map();

	dataMap.forEach((item, key) => {
		for (let i = 0; i < item.length; i++) {
			transformedMap.set(new String(item[i]).toLowerCase(), parseInt(key));
		}
	});

	return Object.fromEntries(transformedMap);
};

File: python/etl/etl.py
def transform(legacy_data: dict[int, list[str]]) -> dict[str, int]:
    transformed_data = {}
    for item in legacy_data:
        for element in legacy_data[item]:
            transformed_data[element.lower()] = item

    return transformed_data        
    #return {legacy_data[item]: item for item in legacy_data}

